python_modules/dagster/dagster/_core/definitions/asset_graph_differ.py

```diff
@@ -28,6 +39,65 @@ class AssetDefinitionChangeType(Enum):
     REMOVED = "REMOVED"
 
 
+class MappedKeyDiff(NamedTuple):
+    added: Set[str]
+    changed: Set[str]
+    removed: Set[str]
+
+
+T = TypeVar("T")
+
+
+@whitelist_for_serdes
+@dataclass(frozen=True)
```

Comment:
We should switch this to record as soon as https://github.com/dagster-io/dagster/pull/24640 is merged